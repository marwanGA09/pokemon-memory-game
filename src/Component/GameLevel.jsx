export function GameLevel({ onGameLevel }) {
  return (
    <div data-testid="game-level" className="main-page">
      <h2>Pokemon Memory Game</h2>
      <p className="description">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe beatae
        recusandae accusamus expedita fuga repellendus deserunt ad molestiae,
        soluta odit dolorum commodi dignissimos nesciunt quo tempore vitae
        ducimus, dolore corrupti.
      </p>
      <p>
        {' '}
        <strong className="play-now">Play Now </strong>
      </p>
      <div>
        <label htmlFor="easy">Easy</label>
        <input
          type="radio"
          value={4}
          name="gameLevel"
          id="easy"
          onChange={(e) => onGameLevel(e.target.value)}
        />
        <label htmlFor="medium">Medium</label>
        <input
          type="radio"
          value={8}
          name="gameLevel"
          id="medium"
          onChange={(e) => onGameLevel(e.target.value)}
        />
        <label htmlFor="hard">Hard</label>
        <input
          type="radio"
          value={12}
          name="gameLevel"
          id="hard"
          onChange={(e) => onGameLevel(e.target.value)}
        />
      </div>
    </div>
  );
}
