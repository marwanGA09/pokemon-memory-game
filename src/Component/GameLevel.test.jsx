import { render, screen } from '@testing-library/react';
import { describe, expect, test, vitest } from 'vitest';
import { GameLevel } from './GameLevel';

const onGameLevelMock = vitest.fn();

describe('GameLevel test', () => {
  test('presence of game level', () => {
    render(<GameLevel onGameLevelMock={onGameLevelMock} />);
    const gameLevelHtml = screen.getByTestId('game-level');
    expect(gameLevelHtml).toBeInTheDocument();
  });

  test('Take snapShot of game level', () => {
    render(<GameLevel onGameLevelMock={onGameLevelMock} />);
    const gameLevelHtml = screen.getByTestId('game-level');
    expect(gameLevelHtml).toMatchSnapshot();
  });

  test('have three level of game', () => {
    render(<GameLevel onGameLevelMock={onGameLevelMock} />);
    const gameLevelInput = screen.getAllByRole('radio');
    screen.debug(gameLevelInput);
    expect(gameLevelInput.length).toBe(3);
  });
});
