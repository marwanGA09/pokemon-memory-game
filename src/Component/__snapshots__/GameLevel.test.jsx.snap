// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameLevel test > Take snapShot of game level 1`] = `
<div
  class="main-page"
  data-testid="game-level"
>
  <h2>
    Pokemon Memory Game
  </h2>
  <p
    class="description"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe beatae recusandae accusamus expedita fuga repellendus deserunt ad molestiae, soluta odit dolorum commodi dignissimos nesciunt quo tempore vitae ducimus, dolore corrupti.
  </p>
  <p>
     
    <strong
      class="play-now"
    >
      Play Now 
    </strong>
  </p>
  <div>
    <label
      for="easy"
    >
      Easy
    </label>
    <input
      id="easy"
      name="gameLevel"
      type="radio"
      value="4"
    />
    <label
      for="medium"
    >
      Medium
    </label>
    <input
      id="medium"
      name="gameLevel"
      type="radio"
      value="8"
    />
    <label
      for="hard"
    >
      Hard
    </label>
    <input
      id="hard"
      name="gameLevel"
      type="radio"
      value="12"
    />
  </div>
</div>
`;

exports[`GameLevel test > have three level of game 1`] = `
<div
  class="main-page"
  data-testid="game-level"
>
  <h2>
    Pokemon Memory Game
  </h2>
  <p
    class="description"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe beatae recusandae accusamus expedita fuga repellendus deserunt ad molestiae, soluta odit dolorum commodi dignissimos nesciunt quo tempore vitae ducimus, dolore corrupti.
  </p>
  <p>
     
    <strong
      class="play-now"
    >
      Play Now 
    </strong>
  </p>
  <div>
    <label
      for="easy"
    >
      Easy
    </label>
    <input
      id="easy"
      name="gameLevel"
      type="radio"
      value="4"
    />
    <label
      for="medium"
    >
      Medium
    </label>
    <input
      id="medium"
      name="gameLevel"
      type="radio"
      value="8"
    />
    <label
      for="hard"
    >
      Hard
    </label>
    <input
      id="hard"
      name="gameLevel"
      type="radio"
      value="12"
    />
  </div>
</div>
`;
